src/Cms/System.php

```diff
@@ -86,6 +86,75 @@ public function curl(): bool
         return extension_loaded('curl');
     }
 
+    /**
+     * Returns the URL to the file within a system folder
+     * if the file is located in the document
+     * root. Otherwise it will return null.
+     *
+     * @param string $folder 'git', 'content', 'site', 'kirby'
+     * @return string|null
+     */
+    public function exposedFileUrl(string $folder): ?string
+    {
+        if (!$url = $this->folderUrl($folder)) {
+            return null;
+        }
+
+        switch ($folder) {
+            case 'content':
+                return $url . '/' . basename($this->app->site()->contentFile());
+            case 'git':
+                return $url . '/config';
+            case 'kirby':
+                return $url . '/composer.json';
+            case 'site':
+                return $url . '/templates/default.php';
```

Comment:
I guess we will ignore the custom `templates` setup as it would be complex?

Reply:
I was wondering about that as well. The default template is such a great place to check because it is pretty much the only thing that's required in the site folder. But with a custom template engine it would be really difficult. Maybe we can fix that later.

Reply:
We always say that and never keep track of this stuff. :)

Possible solution:

1. Check if the `default.php` exists in the actual `templates` root. If yes, check if `templates` is inside the `site` root. If yes, use this path.
2. If `default.php` does not exist, use `scandir` to find any file with the basename `default` (with whatever extension).
3. If the `templates` dir is outside of the `site` dir, ?? Maybe fall back to `config`?

Reply:
We need to make sure though that the "any file" does not leak sensitive information via the API. With a template it would be no problem, but apart from that we need to be careful.

Reply:
You got a point there. Maybe this entire section should only be accessible for admins anyway.

Reply:
I fixed it by only checking for files within blueprints, templates and snippets folders. In combination with permissions, I think we are good here now